embassy-usb/src/builder.rs

```diff
@@ -201,6 +201,20 @@ impl<'d, D: Driver<'d>> Builder<'d, D> {
         self.config_descriptor.end_configuration();
         self.bos_descriptor.end_bos();
 
+        info!("USB: device_descriptor used: {}", self.device_descriptor.position());
+        info!("USB: config_descriptor used: {}", self.config_descriptor.position());
+        info!("USB: bos_descriptor_buf used: {}", self.bos_descriptor.writer.position());
+        #[cfg(feature = "msos-descriptor")]
+        info!("USB: device_descriptor used: {}", msos_descriptor.len());
+        if self.control_buf.len() != self.config.max_packet_size_0.into() {
+            warn!(
```

Comment:
I'm not sure if this should be an error (either the log level, a panic, or otherwise). Let me know if you have thoughts.

Reply:
I'm sorry, I completely brainfarted on Matrix. I told you the `control_buf` size must be equal to the EP0 MPS, but that's not true at all. Control payloads get fragmented into multiple MPS-sized packets, so the buffer size can be anything (usually bigger, but could be smaller too). So this check is not correct :(

Reply:
Will remove that warning!